core/permissions/app_token.go

```diff
@@ -0,0 +1,96 @@
+package permissions
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"strings"
+)
+
+var invalidAppTokenErr = errors.New("app token is invalid")
+
+const tokenKeyLength = 20
+const tokenSecretLength = 30
+
+type AppToken struct {
+	Key         string
+	Secret      string
+	IsMaster    bool
+	Permissions []string
+}
+
+// Token structure is [KEY].[SECRET].[IS_ADMIN].[PERMISSION1]_[PERMISSION2]...
```

Comment:
Perhaps marshalling into a json string might be better and less error prone? Is there a reason why this format was chosen?

Reply:
I can't recall why I went for this approach but you're right, I'll update it to a json marshal.

Reply:
Fixed in 9fe539a